Bullwinkle.class.nut

```diff
@@ -162,11 +165,18 @@ class Bullwinkle {
         if (_watchdogTimer == null) _watchdog();
 
         // Increment the tries
-        if (message.type == BULLWINKLE_MESSAGE_TYPE.SEND) {
-        	message.tries++;
+        if (message.type == BULLWINKLE_MESSAGE_TYPE.SEND && message.id in _packages) {
+		        _packages[message.id]._tries++;
```

Comment:
Is it worth making the breaking change to reduce overhead on the wire? (I would think from imp's perspective for "free" developer devices and your paying customer's perspective for large device deployments, that would be a win...)

Aron - could the onFail callback function be updated to something like 
```
.onFail(function(err, message, retry, numTries) {
        // Try sending the message again in 60 seconds, assuming we've not made more than 3 tries
  if(numTries <3){
        if (!retry(60)) {
            server.error("No more retry attempts are allowed");
        }
  }
})
```

I'm also curious as to why you look at message.tries directly instead of just passing the ```maxRetries``` setting and let the class take care of it for you?  I've not used the retry functionality too much myself (TCP typically takes care of it for me, or I just assume the message will fail and I need to deal with it some other way) so I may be missing something...